storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py

```diff
@@ -238,7 +238,7 @@ def soft_delete_relative_cost_analyzer(
     agg_days: int = 30,
     lookback_days: int = 360,
     list_buckets: bool = False,
-) -> str | Dict[str, float]:  # Note potential string output
+    ) -> Union[str, Dict[str, float]]: # Note potential string output
```

Comment:
Thanks for the PR. As per https://docs.python.org/3/library/typing.html#special-forms, `Union` does the same job as `|`. Any particular reason why change this ?

Reply:
Thanks for quick replay! Running the script with python 3.9 I got this issue:

python-docs-samples/storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py", line 241, in <module>
    ) -> str | Dict[str, float]:  # Note potential string output
TypeError: unsupported operand type(s) for |: 'type' and '_GenericAlias'

The above change fixes this.

Reply:
The shorthand '|' for union is only supported in python 3.10 and up.

Reply:
We can still use `|` in Python 3.9 by adding `from __future__ import annotations`, just confirmed in my local setup.